### PR TITLE
Add check for square matrix input in normestinv

### LINKED ISSUE #12789: `Base.SparseMatrix.normestinv` does not check if input is square
Since generally the inverse of a rectangular matrix does not exist, estimating the norm of the inverse is not meaningful. However, the current implementation of `normestinv` does not check if the input is square, even though the code cannot handle rectangular matrices.
```julia
julia> Base.SparseMatrix.normestinv(sprand(3,5,.9))
ERROR: DimensionMismatch("Q matrix size 3 and dense matrix has 5 rows")
 in qmult at sparse/spqr.jl:130
 in \ at sparse/spqr.jl:144
 in normestinv at sparse/linalg.jl:561
 in normestinv at sparse/linalg.jl:509

julia> Base.SparseMatrix.normestinv(sprand(5,3,.9))
ERROR: BoundsError: attempt to access 3x2 Array{Float64,2}:
 -0.247764   0.793321 
  0.332851   0.0269186
  0.163862  -0.654671 
  at index [1:5,1]
 in throw_boundserror at abstractarray.jl:153
 in normestinv at sparse/linalg.jl:565
 in normestinv at sparse/linalg.jl:509
```

### PR DESCRIPTION
**Pull request by marcusps**
Fixes #12789 (as well as an indentation problem)

### CODE REVIEW

File: base/sparse/linalg.jl [hunk 1]
```diff
@@ -510,7 +510,8 @@ end
 function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = min(2,maximum(size(A))))
     maxiter = 5
     # Check the input
-    n = max(size(A)...)
+    chksquare(A)
```
Comment (andreasnoack) [inline base/sparse/linalg.jl]: `chksquare` returns the size so you can just do `n=chksquare(A)` and remove the next line.

Comment (marcusps) [inline base/sparse/linalg.jl]: Done.

**Comment by andreasnoack** [comment]
Thanks. The changes look good to me now.

**Comment by marcusps** [comment]
Let me push the "eachindex" change @stevengj suggested, and then it should be good to go.

**Pull request merged**